tests/integration/components/nav-route-test.js

```diff
@@ -8,15 +10,129 @@ describe(test.label, function () {
   test.setup()
 
   beforeEach(function () {
-    this.render(hbs`{{nav-route}}`)
+    this.render(hbs`
+      {{nav-route
+        hook='hook'
+        hookQualifiers=(hash fizz='bang')
+        icon=icon
+        route=route
+      }}
+    `)
   })
 
-  it('should render a single element in the DOM', function () {
-    expect(this.$()).to.have.length(1)
+  describe('when route is undefined', function () {
+    beforeEach(function () {
+      this.set('route', undefined)
+      return wait()
+    })
+
+    describe('when icon not a path', function () {
+      beforeEach(function () {
+        this.set('icon', 'add')
+        return wait()
+      })
+
+      it('should have frost-icon class', function () {
+        const $icon = $hook('hook-link-icon', {fizz: 'bang'})
+        expect($icon.attr('class').split(' ')).to.include('frost-icon')
+      })
+
+      it('should have the expected xlink:href', function () {
+        const $icon = $hook('hook-link-icon', {fizz: 'bang'})
+        expect($icon.find('use')).to.have.attr('xlink:href', 'assets/icon-packs/frost.svg#add')
+      })
+    })
+
+    describe('when icon is a path without an id', function () {
+      beforeEach(function () {
+        this.set('icon', '/my/icon/path')
+        return wait()
+      })
+
+      it('should have frost-icon class', function () {
+        const $icon = $hook('hook-link-icon', {fizz: 'bang'})
```

Comment:
The assignment of the jQuery object to `$icon` creates a reference to that object prevents clean up after the test finishes unless the variable is assigned `null`.  If the `$icon` variable assignment is setup in a `beforeEach()` the `null` assignment of the `$icon` variable could be done in an `afterEach()`. Send @notmessenger or myself Slack message and we can point you to an example in one of our apps.

Reply:
I understand exactly what you mean regarding `beforeEach` and `afterEach`, but aren't `const` variables automatically freed when they go out of scope?

Reply:
@sophypal are you sure the leak happens in this case? I could see it happening you define the variable outside of an `it` block and then assign it, but defining it in the `it` block should be safe, no?

Reply:
@TheOtherDude is correct. Closures for variables defined **outside** `it` test descriptions are not freed during test because each `describe` test module is held on by the test runner.

Reply:
That's what I would think too. I thought the problem had to do with declaring a variable with `let` outside of `beforeEach()` and `it()` because the `function` blocks passed into those create closures around the declared variable, and those hang around. I didin't think it would be an issue with a variable declared within a `function` block that has no closures beneath it.

Reply:
I think @juwara0 may have been thinking about this in the context of not having assignments done in `it()`s as per https://github.com/ciena-frost/ember-frost-test#it-blocks.  So if we're okay having them in the `it()`s then the code will be fine but if we follow https://github.com/ciena-frost/ember-frost-test#it-blocks and don't then @juwara0 warnings are apt.

Reply:
Dang, now I look like I was just repeating what @sophypal said, b/c I was too slow with my comment ;)

Reply:
I don't think anything in https://github.com/ciena-frost/ember-frost-test#it-blocks says you shouldn't have assignments. It talks about not performing actions or "doing something" 

Maybe I was unclear when I wrote that, but I wasn't considering fetching something and storing it in a variable to reference inside an `expect()` as "doing something" 

Without that assignment, you end up with:

```js
expect($hook('hook-link-icon', {fizz: 'bang'}).attr('class').split(' ')).to.include('frost-icon')
```

which I think is a bit harder to follow.